src/wallet/scriptpubkeyman.cpp

```diff
@@ -1717,42 +1717,190 @@ std::unordered_set<CScript, SaltedSipHasher> LegacyDataSPKM::GetScriptPubKeys()
+        // P2PK in any nesting level is valid, but inside of P2WSH, the pubkey must also be compressed.
+        case TxoutType::PUBKEY:
+            if (ctx == ScriptContext::P2WSH && solutions[0].size() != CPubKey::COMPRESSED_SIZE) return false;
+            return true;
+        // P2WPKH is allowed as an output script or in P2SH, but not inside of P2WSH
+        case TxoutType::WITNESS_V0_KEYHASH:
+            return ctx != ScriptContext::P2WSH;
+        // P2PKH in any nesting level is valid, but inside of P2WSH, the pubkey must also be compressed.
+        // If the pubkey cannot be retrieved to check for compression, then the P2WSH-P2PKH is allowed.
+        case TxoutType::PUBKEYHASH:
+            if (ctx == ScriptContext::P2WSH) {
+                CPubKey pubkey;
+                if (GetPubKey(CKeyID(uint160(solutions[0])), pubkey) && !pubkey.IsCompressed()) {
+                    return false;
+                }
+            }
+            return true;
+        // P2SH is allowed only as an output script.
+        // If the redeemScript is known, it must also be a valid script within P2SH.
+        // If the redeemScript is not known, the P2SH output script is valid.
+        case TxoutType::SCRIPTHASH:
+        {
+            if (ctx != ScriptContext::TOP) return false;
+            CScriptID script_id = CScriptID(uint160(solutions[0]));
+            CScript subscript;
+            if (GetCScript(script_id, subscript)) {
+                return is_valid_script(subscript, ScriptContext::P2SH);
+            }
+            return true;
+        }
+        // P2WSH is allowed as an output script or inside of P2SH, but not P2WSH.
+        // If the witnessScript is known, it must also be a valid script within P2WSH.
+        // If the witnessScript is not known, the P2WSH output script is valid.
+        case TxoutType::WITNESS_V0_SCRIPTHASH:
+        {
+            if (ctx == ScriptContext::P2WSH) return false;
+            // CScriptID is the hash160 of the script. For P2WSH, we already have the SHA256 of the script,
+            // so only RIPEMD160 of that is required to get the CScriptID for lookup.
+            CScriptID script_id{RIPEMD160(solutions[0])};
+            CScript subscript;
+            if (GetCScript(script_id, subscript)) {
+                return is_valid_script(subscript, ScriptContext::P2WSH);
+            }
+            return true;
+        }
+        // Multisig in any nesting level is valid, but inside of P2WSH, all pubkeys must be compressed.
+        case TxoutType::MULTISIG:
+        {
+            if (ctx == ScriptContext::P2WSH) {
+                std::vector<std::vector<unsigned char>> keys(solutions.begin() + 1, solutions.begin() + solutions.size() - 1);
+                if (!all_keys_compressed(keys)) {
+                    return false;
+                }
+            }
+            return true;
+        }
+        }
+        assert(false);
+    };
+    // Iterate again for all the P2WSH scripts
```

Comment:
I’m pretty lost at this point. We already handled multisig and even really weird stuff like putting a PKH into a SH. What exactly are we looking for here that we then not also skip?

Reply:
This was to handle pkh inside of wsh, which was a previously unhandled case. Furthermore, I thought that it would be possible for other (nonstandard) scripts to be included in here, however further analysis shows that this is not possible as nonstandard scripts are always `ISMINE_NO`.

I've removed this loop and added wsh handling for pkh to the loop above.